utils/kafkaclient/config.go

```diff
@@ -0,0 +1,124 @@
+package kafkaclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"time"
+
+	"github.com/segmentio/kafka-go/sasl"
+	"github.com/segmentio/kafka-go/sasl/plain"
+	"github.com/segmentio/kafka-go/sasl/scram"
+)
+
+type ScramHashGenerator uint8
+
+const (
+	ScramPlainText ScramHashGenerator = iota
+	ScramSHA256
+	ScramSHA512
+)
+
+func (s ScramHashGenerator) String() string {
+	switch s {
+	case ScramPlainText:
+		return "plain"
+	case ScramSHA256:
+		return "sha-256"
+	case ScramSHA512:
+		return "sha-512"
+	default:
+		panic(fmt.Errorf("scram hash generator out of the known domain %d", s))
+	}
+}
+
+type Config struct {
+	ClientID    string
+	DialTimeout time.Duration
+	TLS         *TLS
```

Comment:
The type of configuration that we support appears to be very specific:
https://www.rudderstack.com/docs/destinations/streaming-platforms/kafka/#apache-kafka

e.g. we don't appear to be supporting client certificates, only a CA certificate for SSL and SASL when SSL is enabled.

Reply:
It will be the responsibility of the `kafkamanager` to use only the proper TLS configuration :+1: This will live there anyway as subpackage as agreed.